### ISSUE TITLE
Fails building with catkin tools

### ISSUE THREAD

**Issue by KevinBollore** [InstitutMaupertuis/fanuc_motion#2]
```
Abandoned <<< fanuc_motion       [ Depends on unknown jobs: fanuc, fanuc_experimental ]
```

**Comment by VictorLamoine** [comment]
I confirm; this happens when `fanuc` and `fanuc_experimental` repositories are in the workspace.

Builds fine with `catkin_make`

**Comment by gavanderhoorn** [comment]
Is this related to ros-industrial/fanuc#185, or is this something else?

**Comment by VictorLamoine** [comment]
I don't think it is.
Probably an issue in our `CMakeLists.txt` / `package.xml` but I don't know what is wrong yet.

**Comment by VictorLamoine** [comment]
You're right, it's failing because of the `fanuc` package!
Nice catch, thanks for saving us debug time.

**Comment by gavanderhoorn** [comment]
I just built `fanuc`, `fanuc_experimental` and some other pkgs in a workspace with `catkin_tools` v0.4.2 and I don't get any errors.

Not saying you're wrong, but I'd be interested to know what is going on.

**Comment by VictorLamoine** [comment]
Can you try with `fanuc_motion` ?
I'm using `catkin_tools` v0.4.2 too.

**Comment by VictorLamoine** [comment]
Full log
```
dell@M6700:~/catkin_test$ ls src/
CMakeLists.txt  fanuc  fanuc_motion
dell@M6700:~/catkin_test$ catkin clean -y
[clean] Removing develspace: /home/dell/catkin_test/devel
[clean] Removing buildspace: /home/dell/catkin_test/build
[clean] Removing log space: /home/dell/catkin_test/logs
dell@M6700:~/catkin_test$ catkin build
---------------------------------------------------------------------------------------
Profile:                     default
Extending:             [env] /home/dell/MEGAsync/catkin_workspace/devel:/opt/ros/indigo
Workspace:                   /home/dell/catkin_test
---------------------------------------------------------------------------------------
Source Space:       [exists] /home/dell/catkin_test/src
Log Space:         [missing] /home/dell/catkin_test/logs
Build Space:        [exists] /home/dell/catkin_test/build
Devel Space:        [exists] /home/dell/catkin_test/devel
Install Space:      [unused] /home/dell/catkin_test/install
DESTDIR:            [unused] None
---------------------------------------------------------------------------------------
Devel Space Layout:          linked
Install Space Layout:        None
---------------------------------------------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
---------------------------------------------------------------------------------------
Whitelisted Packages:        None
Blacklisted Packages:        None
---------------------------------------------------------------------------------------
Workspace configuration appears valid.

NOTE: Forcing CMake to run for each package.
---------------------------------------------------------------------------------------
[build] Found '23' packages in 0.0 seconds.                                                                                                                                            
[build] Updating package table.                                                                                                                                                        
Abandoned <<< fanuc_motion                                     [ Depends on unknown jobs: fanuc ]                                                                                      
Starting  >>> catkin_tools_prebuild                                                                                                                                                    
Finished  <<< catkin_tools_prebuild                            [ 1.3 seconds ]                                                                                                         
Starting  >>> fanuc_driver                                                                                                                                                             
Starting  >>> fanuc_lrmate200ic_moveit_plugins                                                                                                                                         
Starting  >>> fanuc_m10ia_moveit_plugins                                                                                                                                               
Starting  >>> fanuc_m16ib_moveit_plugins                                                                                                                                               
Starting  >>> fanuc_m20ia_moveit_plugins                                                                                                                                               
Starting  >>> fanuc_m430ia_moveit_plugins                                                                                                                                              
Starting  >>> fanuc_resources                                                                                                                                                          
Finished  <<< fanuc_resources                                  [ 1.8 seconds ]                                                                                                         
Finished  <<< fanuc_driver                                     [ 3.2 seconds ]                                                                                                         
Starting  >>> fanuc_lrmate200ic_support                                                                                                                                                
Starting  >>> fanuc_m10ia_support                                                                                                                                                      
Finished  <<< fanuc_m20ia_moveit_plugins                       [ 3.3 seconds ]                                                                                                         
Starting  >>> fanuc_m16ib_support                                                                                                                                                      
Starting  >>> fanuc_m20ia_support                                                                                                                                                      
Finished  <<< fanuc_m10ia_moveit_plugins                       [ 3.4 seconds ]                                                                                                         
Finished  <<< fanuc_m16ib_moveit_plugins                       [ 3.6 seconds ]                                                                                                         
Finished  <<< fanuc_m430ia_moveit_plugins                      [ 3.6 seconds ]                                                                                                         
Starting  >>> fanuc_m430ia_support                                                                                                                                                     
Finished  <<< fanuc_lrmate200ic_moveit_plugins                 [ 3.8 seconds ]                                                                                                         
Finished  <<< fanuc_m10ia_support                              [ 1.6 seconds ]                                                                                                         
Starting  >>> fanuc_m10ia_moveit_config                                                                                                                                                
Finished  <<< fanuc_lrmate200ic_support                        [ 1.6 seconds ]                                                                                                         
Starting  >>> fanuc_lrmate200ic5h_moveit_config                                                                                                                                        
Starting  >>> fanuc_lrmate200ic5l_moveit_config                                                                                                                                        
Starting  >>> fanuc_lrmate200ic_moveit_config                                                                                                                                          
Finished  <<< fanuc_m16ib_support                              [ 1.7 seconds ]                                                                                                         
Starting  >>> fanuc_m16ib20_moveit_config                                                                                                                                              
Finished  <<< fanuc_m20ia_support                              [ 1.7 seconds ]                                                                                                         
Starting  >>> fanuc_m20ia10l_moveit_config                                                                                                                                             
Starting  >>> fanuc_m20ia_moveit_config                                                                                                                                                
Finished  <<< fanuc_m430ia_support                             [ 1.3 seconds ]                                                                                                         
Starting  >>> fanuc_m430ia2f_moveit_config                                                                                                                                             
Finished  <<< fanuc_m10ia_moveit_config                        [ 1.5 seconds ]                                                                                                         
Starting  >>> fanuc_m430ia2p_moveit_config                                                                                                                                             
Finished  <<< fanuc_lrmate200ic5l_moveit_config                [ 1.7 seconds ]                                                                                                         
Finished  <<< fanuc_lrmate200ic5h_moveit_config                [ 1.7 seconds ]                                                                                                         
Finished  <<< fanuc_lrmate200ic_moveit_config                  [ 1.7 seconds ]                                                                                                         
Finished  <<< fanuc_m20ia_moveit_config                        [ 1.7 seconds ]                                                                                                         
Finished  <<< fanuc_m16ib20_moveit_config                      [ 1.9 seconds ]                                                                                                         
Finished  <<< fanuc_m20ia10l_moveit_config                     [ 1.8 seconds ]                                                                                                         
Finished  <<< fanuc_m430ia2f_moveit_config                     [ 1.5 seconds ]                                                                                                         
Finished  <<< fanuc_m430ia2p_moveit_config                     [ 1.2 seconds ]                                                                                                         
[build] Summary: 22 of 23 packages succeeded.                                                                                                                                          
[build]   Ignored:   1 packages were skipped or are blacklisted.                                                                                                                       
[build]   Warnings:  None.                                                                                                                                                             
[build]   Abandoned: 1 packages were abandoned.                                                                                                                                        
[build]   Failed:    None.                                                                                                                                                             
[build] Runtime: 9.5 seconds total.                                                                                                                                                    
[build] Note: Workspace packages have changed, please re-source setup files to use them.
```

**Comment by gavanderhoorn** [comment]
My workspace has `industrial_core` as a source package, and I get:
```shell
Abandoned <<< fanuc_motion  [ Depends on unknown jobs: fanuc, fanuc_experimental, industrial_core ]
```
as soon as I add `fanuc_motion`.

Not sure what is going on.

**Comment by VictorLamoine** [comment]
Are those exec tags ok?
https://github.com/InstitutMaupertuis/fanuc_motion/blob/indigo-devel/package.xml#L25-L26

**Comment by gavanderhoorn** [comment]
I think it's something to do with the `exec_depend`s in your package manifest.

As soon as I comment the `fanuc`, `fanuc_experimental` and `industrial_core` depends, the message goes away.

**Comment by VictorLamoine** [comment]
So how should I add a dependency on `fanuc` / `fanuc_experimental` ?
It's not a build dependency because the package can be built without those packages, it's a run-time dependency.

**Comment by gavanderhoorn** [comment]
Based on [REP-140 - Data Representation - <exec_depend> (multiple)](http://www.ros.org/reps/rep-0140.html#exec-depend-multiple) I'd say those are fine:

> The `<exec_depend>` is needed for packages providing shared libraries, executable commands, Python modules, launch scripts or any other files required for running your package. It is also used by metapackages for grouping packages.

Might be something to ask the guys over at [catkin/catkin_tools](https://github.com/catkin/catkin_tools/).

**Comment by gavanderhoorn** [comment]
Building with CMI seems to succeed. `fanuc_motion` builds.

**Comment by VictorLamoine** [comment]
Fixed with ba2b9ddd054b3ec0ff488657c291f6a2348e969b
